test/printf.c

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
```

Comment:
Talking with @dschuff, this test probably shouldn't actually be relegated to wasm_backend only. The reason being there's no guarantee that we'll never have truncated stores in asm2wasm, so as soon as we do this case is being generated and not tested.

The reason the wasm backend emits i64.store32 is simply to avoid two operations where one would suffice. It would be possible (and maybe marginally beneficial?) to add a peephole pass in binaryen that converts `(i32.store (i32.wrap <i64 expr>))` to `(i64.store32 <i64 expr>)`

The C code in `vprintf.c` that winds up generating the i64.store8s looks like:
```c
static char *fmt_u(uintmax_t x, char *s)
{
	unsigned long y;
	for (   ; x>ULONG_MAX; x/=10) *--s = '0' + x%10;
	for (y=x;           y; y/=10) *--s = '0' + y%10;
	return s;
}
```
(uintmax_t is i64 on 64-bit targets.)
So we have an i64 and write to an i8, so i64.store8 is a natural instruction.